### ISSUE TITLE
[Question] Resuming training of normalized environment decreases dramatically

### ISSUE THREAD

**Issue by mindacrobatic** [DLR-RM/stable-baselines3#1256]
### ❓ Question

Hello everybody!

I have trained Humanoid-v3 in Google Colabs and the average episode reward looked really good. But when I want to test the agent locally, it does not perform well at all. And resuming the training on my local computer shows a HUGE decrease in average episode reward 4000 down to 1000.

I think I did it just like in this example: [Pybullet with normalized features](https://stable-baselines3.readthedocs.io/en/master/guide/examples.html#pybullet-normalizing-input-features)

To show you the most important parts:

In Colab, I trained the agent this way:
```python
env = SubprocVecEnv([lambda: gym.make('Humanoid-v3')]*16)
env = VecNormalize(env, norm_obs=True, norm_reward=False)
env = VecMonitor(env)

# Save a checkpoint every 10000 steps
checkpoint_callback = CheckpointCallback(
    save_freq=10000,
    save_path="drive/MyDrive/Humanoid/checkpoints",
    save_vecnormalize=True
)

# Learning rate scheduler
def linear_schedule(initial_value: float):
    def func(progress_remaining: float):
        return progress_remaining * initial_value
    return func

model = PPO(
    "MlpPolicy", 
    env,
    learning_rate=0.0001,
    verbose=0,
    tensorboard_log="drive/MyDrive/Humanoid/tensorboard/"
)

model.learn(
    total_timesteps=10_000_000,
    callback=checkpoint_callback
)
```

When I resumed the training on my local computer, I did the following:

```python
env = SubprocVecEnv([lambda: gym.make('Humanoid-v3')]*16)
env = VecNormalize.load("checkpoints/rl_model_vecnormalize_10000000_steps.pkl", env)
env = VecMonitor(env)

env.norm_obs = True
env.norm_reward = False
env.training = True

model = PPO.load(
    "checkpoints/rl_model_10000000_steps.zip",
    env=env,
    custom_objects={
        "tensorboard_log": "tensorboard"
    }
)

model.learn(
    total_timesteps=2_000_000,
    reset_num_timesteps=False,
    callback=checkpoint_callback
)
```

This is, how the curve of my average episode looks like after resuming:
<img width="659" alt="Learning decrease" src="https://user-images.githubusercontent.com/90468068/210134070-9c0a3c62-b9a8-45dc-a83b-bcafc26ddae5.png">

Does anybody have an idea what is going wrong here?

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
thanks for reporting the issue.
I tried to reproduce with a shorter experiment but couldn't so far.

The only thing suspicious to me is the learning rate schedule. Are you using one or not? (you define it but it is not used)

train:
```python
from stable_baselines3 import PPO
from stable_baselines3.common.vec_env import SubprocVecEnv, VecNormalize
from stable_baselines3.common.env_util import make_vec_env
from stable_baselines3.common.callbacks import CheckpointCallback

if __name__ == "__main__":
    env_id = "LunarLander-v2"
    n_envs = 4
    env = make_vec_env(env_id, n_envs=4, vec_env_cls=SubprocVecEnv)
    env = VecNormalize(env, norm_obs=True, norm_reward=False)

    # Save a checkpoint every 10000 steps
    checkpoint_callback = CheckpointCallback(
        save_freq=50000 // n_envs, save_path="logs/checkpoints", save_vecnormalize=True
    )

    model = PPO(
        "MlpPolicy",
        env,
        learning_rate=0.0001,
        verbose=0,
        tensorboard_log="logs/tensorboard/",
        seed=1,
    )

    model.learn(
        total_timesteps=int(2e5),
        callback=checkpoint_callback,
        progress_bar=True,
        tb_log_name=f"ppo_{env_id}",
    )
    model.save(f"logs/ppo_{env_id}")
    env.save(f"logs/ppo_vecnormalize_{env_id}.pkl")

```

continue training:
```python
from stable_baselines3 import PPO
from stable_baselines3.common.vec_env import SubprocVecEnv, VecNormalize, VecMonitor
from stable_baselines3.common.env_util import make_vec_env

if __name__ == "__main__":
    env_id = "LunarLander-v2"
    env = make_vec_env(env_id, n_envs=4, vec_env_cls=SubprocVecEnv)

    # env = VecNormalize.load(
    #     "logs/checkpoints/rl_model_vecnormalize_100000_steps.pkl", env
    # )
    env = VecNormalize.load("logs/ppo_vecnormalize_LunarLander-v2.pkl", env)

    model = PPO.load(
        # "logs/checkpoints/rl_model_100000_steps.zip",
        "logs/ppo_LunarLander-v2.zip",
        env=env,
    )

    model.learn(
        total_timesteps=int(1e5),
        reset_num_timesteps=False,
        progress_bar=True,
        tb_log_name=f"ppo_{env_id}",
    )

```

Yellow curve is 2e5 training steps at once, blue curve is 1e5 then 1e5.

![Capture d’écran du 2023-01-03 14-45-53](https://user-images.githubusercontent.com/1973948/210369441-62e395f2-a4d3-4ba2-9cc7-a4864b258d07.png)

**Comment by tanielsfranklin** [comment]
I'm facing the same problem with my custom env, using PPO, and VecNormalize.load.
The value loss drops deeply when I use VecNormalize.load.
So I decided to remove VecNormalize.load and now the model takes some time to recover the previous stats 

![Screenshot from 2024-03-26 11-35-28](https://github.com/DLR-RM/stable-baselines3/assets/130485125/d20f766e-6f13-4a34-a0e3-14a4fa72a4a7)
![Screenshot from 2024-03-26 11-36-44](https://github.com/DLR-RM/stable-baselines3/assets/130485125/32698e16-ce31-4034-beca-054f7d33e2dc)

**Comment by tanielsfranklin** [comment]
> I'm facing the same problem with my custom env, using PPO, and VecNormalize.load. The value loss drops deeply when I use VecNormalize.load. So I decided to remove VecNormalize.load and now the model takes some time to recover the previous stats
> 
> ![Screenshot from 2024-03-26 11-35-28](https://private-user-images.githubusercontent.com/130485125/316905888-d20f766e-6f13-4a34-a0e3-14a4fa72a4a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE0ODA4MzUsIm5iZiI6MTcxMTQ4MDUzNSwicGF0aCI6Ii8xMzA0ODUxMjUvMzE2OTA1ODg4LWQyMGY3NjZlLTZmMTMtNGEzNC1hMGUzLTE0YTRmYTcyYTRhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNlQxOTE1MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMWY2YmJmODczMDE3ZWM1YTU2OWQyNGFjODhlZDEyNzFmNjFiYzBkMTFiZjM2NmY4MmM4NzU5N2JmZmQ4YjI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3PO8UnlVKERMkfY-Mqx_AtFf9GyX6NzLPy77X7zj_qA) ![Screenshot from 2024-03-26 11-36-44](https://private-user-images.githubusercontent.com/130485125/316906265-32698e16-ce31-4034-beca-054f7d33e2dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE0ODA4MzUsIm5iZiI6MTcxMTQ4MDUzNSwicGF0aCI6Ii8xMzA0ODUxMjUvMzE2OTA2MjY1LTMyNjk4ZTE2LWNlMzEtNDAzNC1iZWNhLTA1NGY3ZDMzZTJkYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNlQxOTE1MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MzI5M2UzYzNmM2NhYTM5OTM4Y2ZmNWM0M2NiZTg1ZWQwMzhiNWIzMjllZmJhNGUzZjRlZmE3N2M5N2Q2NTZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4Au9dXyW1N1KrTRUdVUX-fZKqCzEe5fMfRyBagjVtWw)

My bad, I was passing a normalized env to the VecNormalized.load.
